jax/lax.py

```diff
@@ -561,14 +565,12 @@ def tan(x):
   return div(sin(x), cos(x))
 
 def asin(x):
-  # asin(x) = 2 * atan(x / (1 + sqrt(1 - x**2)))
-  return mul(_const(x, 2.),
-             atan2(x, add(_const(x, 1.), sqrt(add(_const(x, 1.), square(x))))))
+  return mul(_const(x, 2),
+             atan2(x, add(_const(x, 1), sqrt(sub(_const(x, 1), square(x))))))
```

Comment:
I note you are inconsistent about 1 vs 1. in this PR.

Reply:
I note that you note it! And I'll fix it too :)